grafana-dashboard.tf

```diff
@@ -0,0 +1,36 @@
+resource "random_integer" "graph_id" {
+  min     = 1000
+  max     = 999999
+  count = length(local.apiary_data_buckets)
+}
+
+data "template_file" "graphs" {
+  template = file("${path.module}/templates/graph.tpl")
+  count = length(local.apiary_data_buckets)
+  vars = {
+    bucket_name = local.apiary_data_buckets[count.index]
+    title_bucket_name = local.apiary_managed_schema_names_replaced[count.index]
+    graph_id = random_integer.graph_id.*.result[count.index]
+  }
+}
+
+data "template_file" "dashboard" {
+  template = file("${path.module}/templates/dashboards.tpl")
+  vars = {
+    panels = "[${join(",", data.template_file.graphs.*.rendered)}]"
+  }
+}
+
+resource "kubernetes_config_map" "grafana-dashboard" {
```

Comment:
How/where is this config map used?  I am not seeing the linkage.

Reply:
Raj would be the best person to ask this. I've pieced this solution together using other examples provided, but could never actually find how the config map is being consumed. I would like to answer this question as well!

Reply:
@rpoluri - can you comment here?  Where does this config map get used and what piece of code uses it?

Reply:
To add for future reference what we discovered: 

linkage is here: https://github.com/helm/charts/blob/master/stable/grafana/values.yaml#L447 and https://github.com/zpor/apiary-data-lake/blob/egdl-308-add-dashboard/grafana-dashboard.tf#L29 - label matches sidecar-configured label, so sidecar auto-reads it